src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/ComponentIdentifier.cs

```diff
@@ -3,13 +3,15 @@
 
 using System;
 using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.CodeAnalysis;
 
 namespace Microsoft.DotNet.UpgradeAssistant.MSBuild
 {
     public class ComponentIdentifier : IComponentIdentifier
     {
-        public ProjectComponents GetComponents(IProject project)
+        public async ValueTask<ProjectComponents> GetComponents(IProject project, CancellationToken token)
```

Comment:
Just wanted to confirm.. 

Is the intent here to use ValueTask to describe an asynchronous return result that we expect to be synchronous most of the time?

Reply:
yup